File: day01/part2.py
with open("input.txt") as f:
    lines = f.read().strip().split("\n")

dial = 50
zero_count = 0

for line in lines:
    if line[0] == "L":
        dir = -1
    else:
        dir = 1
    amount = int(line[1:])
    for _ in range(amount):
        dial = (dial + dir + 100) % 100
        if dial == 0:
            zero_count += 1

print("Answer:", zero_count)

File: day01/part2.js
const fs = require("fs");

const lines = fs.readFileSync("input.txt", "utf-8").trim().split("\n");

let dial = 50;
let zero_count = 0;

for (const line of lines) {
  const dir = line[0] === "L" ? -1 : 1;
  const amount = parseInt(line.slice(1), 10);
  for (let i = 0; i < amount; i++) {
    dial = (dial + dir + 100) % 100;
    if (dial === 0) {
      zero_count += 1;
    }
  }
}

console.log("Answer", zero_count);
